transaction.go

```diff
@@ -0,0 +1,158 @@
+package seth
+
+import (
+	"bytes"
+	"math/big"
+)
+
+type Encoding int
+
+const (
+	LIST Encoding = iota
+	STRING
+	INTEGER
+)
+
+type SignedTransaction struct {
```

Comment:
It seems like this could be combined into the `Transaction` type, with a nil `Signature` member indicating that it's not signed. You've probably thought about this more than I have, so if you think this is a more natural or clean way to express it, I'm fine with it.